cornell-dti/office-hours
# Added time limit for questions

### Summary 
Adding a time limit option that professors and TAs can enable so that a sound and banner will display at a warning time and at the time limit time for each question, signifying that the TA should move on to the next question.



I added an option in the professor dashboard under the settings button to add a time limit as well as a time warning for questions, such that at those times, a banner will display and a sound will play, letting the TA and student know that the time for their question is almost up or finished. This can be disabled in case the professor does not want to use this.





### Test Plan 
- Go to the professor view and go to "settings" and enter a time limit as well as a time warning for a class
- Add a question as a student and assign yourself as a TA, and wait to see if you get the banner and sound for the time warning you specified as well as the time limit
- Due to bugs and also just because it was something I wanted to add, make sure that a student can't remove their question when they are assigned
- Make sure that if a TA unassigns himself or herself, as well as marks as done or no show a question, the banner as well as the sound is not sent
- Try disabling the time limit and time warning with the checkbox and see if the banner and sound are still sent


### Notes 



### Breaking Changes 

I updated existing types in `/src/components/types/`
